app/src/main/java/org/mozilla/webmaker/web/javascript/WebAppInterface.java

```diff
@@ -254,12 +253,12 @@ public String getRouteParams() {
 
     @JavascriptInterface
     public String getRouteData() {
-        return MemStorage.sharedStorage().get(ROUTE_KEY);
+        return api.getPayloads(ROUTE_KEY);
     }
 
     @JavascriptInterface
     public void clearRouteData() {
-        MemStorage.sharedStorage().put(ROUTE_KEY, "");
+        api.queue(ROUTE_KEY, "");
```

Comment:
there is a clear method already, so this can be turned into `api.clearPayloads(ROUTE_KEY)`